test-exec/testSender.cpp

```diff
@@ -0,0 +1,18 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+#include<iostream>
+#include <stdlib.h>
+#include <sstream>
+#include "../sender/bms_sender.h"
+#include "../test-exec/catch.hpp"
+
+TEST_CASE("data generation test") {
+
+	BMS_generator_tester * tester_obj = new BMS_generator_tester;
+	BMS_DATA testData;
+	for (int noOfIter = 0; noOfIter < 50; noOfIter++) 
+	{
+		testData = tester_obj->pushToTest(tester_obj->Output_JSON_Data());
+		REQUIRE(tester_obj->Validate_Range(testData.SOC, MIN_SOC, MAX_SOC));

```

Comment:
These tests are checking a random generator. Even if the limits are wrong in the code, it can pass. Moreover, the random generator is a platform functionality, proven to work. You dont need to test it.
There is no test to check the output format - that is a crucial part of your functionality. It also helps future maintainers to catch errors themselves, instead of relying on a receiver to show the error for them.